### ISSUE TITLE
[Mac] - Initial launch begins with old cached app

### ISSUE THREAD

**Issue by matthew-carroll** [flutter/flutter#151578]
### Steps to reproduce

Build/launch a Flutter app on Mac.

### Expected results

The launched app reflects the current code in the app project.

### Actual results

The launched app runs old code from some earlier time.

Running Hot Restart immediately after launch correctly updates the app to reflect the latest code.

### Code sample

<details open><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

https://github.com/flutter/flutter/assets/7259036/5d0e4be6-f96b-4349-8545-f764be96305e

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
Flutter 3.22.2 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 761747bfc5 (5 weeks ago) • 2024-06-05 22:15:13 +0200
Engine • revision edd8546116
Tools • Dart 3.4.3 • DevTools 2.34.3
```

</details>

**Comment by huycozy** [comment]
Hi @matthew-carroll
I wonder if your previous and last code is Dart only or did you have native code changes (macOS) as well? I tried reproducing this case by following steps but couldn't reproduce it:
1. Run a simple app which has a TextField only
2. Terminate it
3. Update Dart code (update decoration for TextField for example)
4. Run app again

Do you see this issue all the time or does it happen randomly?

**Comment by matthew-carroll** [comment]
> I wonder if your previous and last code is Dart only or did you have native code changes (macOS) as well?

I'm not entirely sure what you're asking, but as you can see in the video, I launched the app, and what was launched didn't have my existing Dart code UI.

> Do you see this issue all the time or does it happen randomly?

It's probably not 100% reproducible, but it is happening A LOT across a bunch of different Mac desktop projects that I have.

**Comment by huycozy** [comment]
>I'm not entirely sure what you're asking, but as you can see in the video, I launched the app, and what was launched didn't have my existing Dart code UI.

Yes, I watched the demo video but I'm unsure if you have modified Dart code only or also native code (Swift) on macos module as well between app launches so that I can try reproducing the issue to make the issue more clearly.

Labeling the issue for insights from macOS platform team.

**Comment by matthew-carroll** [comment]
I did make some changes to the Mac window, but that's all. For example I missed with the title bar and the window background color. But I didn't touch any structural elements, nor did I alter any build processes or configuration.

**Comment by loic-sharma** [comment]
Thanks for opening this issue. Unfortunately, the macOS team doesn't have capacity to investigate at this time. Please let us know if you find repro steps, this will make it easier for us to fix this.
